### PR TITLE
DOC-5731: ADVISOR function

### PR DESCRIPTION
**Pull request by simon-dew**
The following documentation is ready for review:

* [ADVISOR Function](https://simon-dew.github.io/docs-site/DOC-5716/server/6.5/n1ql/n1ql-language-reference/advisor.html)
* [Monitor Queries › system:tasks_cache](https://simon-dew.github.io/docs-site/DOC-5716/server/6.5/manage/monitor/monitoring-n1ql-query.html#sys-tasks-cache)

Documentation issue: [DOC-5731](https://issues.couchbase.com/browse/DOC-5731)

### CODE REVIEW

**Comment by simon-dew** [comment]
**Results** > **current_used_indexes** > `If the query uses an index` — need to reword this as there may be multiple queries if invoked using an array or collection.

**Comment by chang27** [comment]
Hi Simon,
For the system:tasks_cache, could you please add Marco (marco.greco@couchbase.com) to review? He implemented that feature. Document link: https://docs.google.com/document/d/1Uv9UE-IjQqyNkuS6ZMxIzx8U8s7VF-sjqkNCQunmZG8/edit#heading=h.kznkrmn1xu6e
Basically it is a common function that can be used in different scenarios. Index Advisor is the first function to use it.

For the Advisor Function just a few tips below:
1. ADVISOR(list_obj): status[Optional]: If "status" is not specified, we returns "all sessions".
2. For Stop/Abort/Purge: the return value has been changed to empty array instead of null:
[
  {
    "$1": []
  }
]
3. For action "stop":  stop collecting queries and the index analysis will run immediately on the current collected queries automatically. Users do not need to invoke index analysis themselves again.

Thank you Simon for composing the index advisor document so nicely!
Bests,
Chang

**Comment by simon-dew** [comment]
> For the system:tasks_cache, could you please add Marco ([marco.greco@couchbase.com](mailto:marco.greco@couchbase.com)) to review? He implemented that feature.

Done

> 1. ADVISOR(list_obj): status[Optional]: If "status" is not specified, we returns "all sessions".

Done

> 2. For Stop/Abort/Purge: the return value has been changed to empty array instead of null:

Done

> 3. For action "stop":  stop collecting queries and the index analysis will run immediately on the current collected queries automatically. Users do not need to invoke index analysis themselves again.

Done

**Comment by marcogrecopriolo** [comment]
Looks ok to me.
One minor point: the "results" field could also appear for canceled tasks.
The mechanics are complex to explain, but essentially
- each task has an execution and stop function associated with it
- execution kicks in after the delay
- stop kicks in if the task is deleted from system:task_cache in order to perform whatever cleanup necessary
- both execution and stop functions could produce results.

**Comment by simon-dew** [comment]
> One minor point: the "results" field could also appear for canceled tasks.

Done
